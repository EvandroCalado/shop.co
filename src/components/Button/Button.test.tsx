import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { Button } from '.';

describe('<Button />', () => {
  it('should render variant primary', () => {
    render(<Button variant="primary">teste</Button>);

    const button = screen.getByRole('button');

    expect(button).toBeInTheDocument();
    expect(button).toHaveClass('bg-black/95 text-white');
    expect(button).not.toHaveClass('bg-white');
  });

  it('should render variant outline', () => {
    render(<Button variant="outline">teste</Button>);

    const button = screen.getByRole('button');

    expect(button).toBeInTheDocument();
    expect(button).toHaveClass('bg-white text-black/95');
    expect(button).not.toHaveClass('bg-black/95');
  });

  it('should render a snapshot', () => {
    const { container } = render(<Button variant="outline">teste</Button>);

    expect(container).toMatchSnapshot();
  });
});
