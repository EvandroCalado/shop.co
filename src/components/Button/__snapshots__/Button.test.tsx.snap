// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render a snapshot 1`] = `
<div>
  <button
    class="border border-black/95 bg-white text-black/95 hover:brightness-95 rounded-full px-14 py-3 text-sm capitalize transition-all duration-150"
  >
    teste
  </button>
</div>
`;
